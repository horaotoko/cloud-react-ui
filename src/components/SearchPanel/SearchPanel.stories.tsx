import { Story, Meta } from "@storybook/react";
import { SearchPanel } from "./SearchPanel";
import type { SearchPanelProps } from "./SearchPanel";
import { Button, TextField, Checkbox } from "../../components";
import styled from "styled-components";

export default {
  component: SearchPanel,
  title: "components/SearchPanel",
} as Meta;

const Fieldset = styled.fieldset`
  border: 0;
  padding: 0;
`;

const Template: Story<SearchPanelProps> = () => (
  <SearchPanel>
    <form>
      <SearchPanel.Filters>
        <SearchPanel.Row>
          <SearchPanel.Column>
            <div>
              <label htmlFor="text-field-1">Text label</label>
              <TextField id="text-field-1" />
            </div>
          </SearchPanel.Column>
          <SearchPanel.Column>
            <div>
              <label htmlFor="text-field-2">Text label</label>
              <TextField id="text-field-2" />
            </div>
          </SearchPanel.Column>
        </SearchPanel.Row>
        <SearchPanel.Row gap={80}>
          <SearchPanel.Column>
            <Fieldset>
              <legend>Checkbox group</legend>
              <Checkbox>checkbox 1</Checkbox>
              <Checkbox>checkbox 2</Checkbox>
              <Checkbox>checkbox 3</Checkbox>
              <Checkbox>checkbox 4</Checkbox>
              <Checkbox>checkbox 5</Checkbox>
            </Fieldset>
          </SearchPanel.Column>
          <SearchPanel.Column>
            <Fieldset>
              <legend>Checkbox group</legend>
              <Checkbox>checkbox 1</Checkbox>
              <Checkbox>checkbox 2</Checkbox>
              <Checkbox>checkbox 3</Checkbox>
              <Checkbox>checkbox 4</Checkbox>
              <Checkbox>checkbox 5</Checkbox>
            </Fieldset>
          </SearchPanel.Column>
        </SearchPanel.Row>
      </SearchPanel.Filters>
      <SearchPanel.Actions>
        <SearchPanel.ButtonGroup>
          <Button color="settings">クリア</Button>
          <Button color="primary">検索</Button>
        </SearchPanel.ButtonGroup>
      </SearchPanel.Actions>
    </form>
  </SearchPanel>
);

export const Default = Template.bind({});
