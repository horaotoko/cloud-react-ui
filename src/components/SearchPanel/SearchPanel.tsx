import styled, { css } from "styled-components";

const StyledSearchPanel = styled.div``;

export type SearchPanelProps = {
  children?: React.ReactNode;
};

export const SearchPanel = (props: SearchPanelProps) => (
  <StyledSearchPanel></StyledSearchPanel>
);
