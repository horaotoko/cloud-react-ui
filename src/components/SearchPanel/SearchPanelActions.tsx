import styled, { css } from "styled-components";
import { defaultProps } from "../../theme";

export const SearchPanelActions = styled.div`
  width: 100%;
  padding-top: 15px;
  padding-bottom: 15px;

  ${({ theme: { searchPanel } }) =>
    css`
      border-top: 1px solid ${searchPanel.borderColor};
    `}
`;

SearchPanelActions.defaultProps = defaultProps;
SearchPanelActions.displayName = "SearchPanel.Actions";
