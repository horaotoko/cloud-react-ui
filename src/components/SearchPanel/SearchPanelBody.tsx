import styled from "styled-components";

type SearchPanelBodyProps = {
  children?: React.ReactNode;
  width?: string;
  ariaHidden: boolean;
};

export const SearchPanelBody = styled.div.attrs<SearchPanelBodyProps>(
  ({ ariaHidden }) => ({
    "aria-hidden": ariaHidden,
  })
)<SearchPanelBodyProps>`
  width: 100%;
  overflow-y: auto;
  transition: max-height 199ms ease-in-out 0ms;

  /* ToggleアニメーションのためにPanelの "おおよそ" のmax-heightを設定 */
  --max-row-number: 3;
  --height-actions: 64px;
  --height-filters-row: 48px;
  --height-filters-total-gaps: calc(15px * (var(--max-row-number) + 1));
  --height-filters: var(--height-filters-row) * var(--max-row-number) +
    var(--height-filters-total-gaps);
  max-height: calc(var(--height-actions) + var(--height-filters));

  &[aria-hidden="true"] {
    visibility: hidden;
    max-height: 0;
    transition: visibility 299ms linear 299ms, max-height 199ms ease-in-out 0ms;
  }
`;

SearchPanelBody.displayName = "SearchPanel.Body";
