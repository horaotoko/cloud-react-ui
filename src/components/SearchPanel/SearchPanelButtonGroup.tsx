import styled from "styled-components";

export const SearchPanelButtonGroup = styled.div`
  width: 100%;
  display: flex;
  justify-content: center;
  column-gap: 8px;
`;

SearchPanelButtonGroup.displayName = "SearchPanel.ButtonGroup";
