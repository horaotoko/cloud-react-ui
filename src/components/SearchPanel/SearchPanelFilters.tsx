import styled from "styled-components";

export const SearchPanelFilters = styled.div`
  padding-top: 15px;
  padding-bottom: 15px;
`;

SearchPanelFilters.displayName = "SearchPanel.Filters";
