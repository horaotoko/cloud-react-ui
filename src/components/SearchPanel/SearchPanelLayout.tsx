import styled from "styled-components";

export const SearchPanelColumn = styled.div``;

SearchPanelColumn.displayName = "SearchPanel.Column";

export const SearchPanelRow = styled.div`
  width: 100%;
  display: flex;
  flex-wrap: wrap;
  column-gap: 8px;

  & + & {
    margin-top: 15px;
  }
`;

SearchPanelRow.displayName = "SearchPanel.Row";

export type SearchPanelFieldGroupProps = {
  children: React.ReactNode;
  type?: "text" | "checkbox";
};

export const SearchPanelFieldGroup = styled.div.attrs<SearchPanelFieldGroupProps>(
  ({ type = "text" }) => ({
    "data-type": type,
  })
)<SearchPanelFieldGroupProps>`
  display: grid;

  [data-type="text"] {
    column-gap: 8px;
  }
  [data-type="checkbox"] {
    column-gap: 84px;
  }
`;

SearchPanelFieldGroup.displayName = "SearchPanel.FieldGroup";
