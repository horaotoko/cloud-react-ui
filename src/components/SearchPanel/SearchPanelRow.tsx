import styled from "styled-components";

export const SearchPanelRow = styled.div`
  width: 100%;
  & + & {
    margin-top: 15px;
  }
`;

SearchPanelRow.displayName = "SearchPanel.Row";
