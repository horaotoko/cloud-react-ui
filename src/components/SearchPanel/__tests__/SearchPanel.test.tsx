import { render, fireEvent, screen } from '@testing-library/react';
import { SearchPanel } from '../SearchPanel';
import { Button } from '../../../components';

describe('SearchPanel', () => {
  it('default', () => {
    const { asFragment } = render(
      <SearchPanel>
        <form>
          <SearchPanel.Filters>
            <SearchPanel.Row>
              <SearchPanel.Column>
                <div>Content</div>
              </SearchPanel.Column>
            </SearchPanel.Row>
          </SearchPanel.Filters>
          <SearchPanel.Actions>
            <SearchPanel.ButtonGroup>
              <Button color="settings">クリア</Button>
              <Button color="primary">検索</Button>
            </SearchPanel.ButtonGroup>
          </SearchPanel.Actions>
        </form>
      </SearchPanel>
    );
    expect(asFragment()).toMatchSnapshot();
    expect(screen.getByText('Content')).not.toBeVisible();
  });

  it("should be open when 'defaultOpen' has true", () => {
    render(<SearchPanel defaultOpen={true}>Content</SearchPanel>);
    expect(screen.getByText('Content')).toBeVisible();
  });

  it('onClick Toggle', () => {
    render(<SearchPanel>Content</SearchPanel>);
    const button = screen.getByRole('button', { expanded: false });

    fireEvent.click(button);
    expect(button).toHaveAttribute('aria-expanded', 'true');
    expect(screen.queryByText('Content')).toBeVisible();

    fireEvent.click(button);
    expect(button).toHaveAttribute('aria-expanded', 'false');
    expect(screen.queryByText('Content')).not.toBeVisible();
  });
});
