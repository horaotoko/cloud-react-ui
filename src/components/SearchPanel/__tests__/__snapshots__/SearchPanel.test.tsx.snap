// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchPanel default 1`] = `
<DocumentFragment>
  .c11 {
  font-size: 13px;
  color: #333333;
}

.c11:hover {
  color: #333333;
}

.c8.c8 {
  display: inline-block;
  height: 32px;
  padding: 0 12px;
  line-height: 32px;
  font-size: 14px;
  border-radius: 4px;
  border: 1px solid #D4D8DD;
  color: #3B7DE9;
  background: transparent;
  cursor: pointer;
  position: relative;
  z-index: 0;
}

.c8.c8:before,
.c8.c8:after {
  content: "";
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  border-radius: 4px;
}

.c8.c8:before {
  z-index: -1;
  background: linear-gradient(to bottom,#FFFFFF,#ECF2FD);
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

.c8.c8:after {
  content: "";
  z-index: -2;
  background: linear-gradient(to bottom,#ECF2FD,#FFFFFF);
  border-radius: 4px;
}

.c8.c8:hover:before {
  opacity: 0;
}

.c8.c8:active,
.c8.c8:focus,
.c8.c8:visited {
  outline: 0;
  box-shadow: 0 0 2px rgba(212,216,221,0.3);
}

.c9.c9 {
  display: inline-block;
  height: 32px;
  padding: 0 12px;
  line-height: 32px;
  font-size: 14px;
  border-radius: 4px;
  border: 1px solid rgba(0,0,0,0.15);
  color: #FFFFFF;
  background: transparent;
  cursor: pointer;
  position: relative;
  z-index: 0;
}

.c9.c9:before,
.c9.c9:after {
  content: "";
  width: 100%;
  height: 100%;
  position: absolute;
  left: 0;
  border-radius: 4px;
}

.c9.c9:before {
  z-index: -1;
  background: linear-gradient(to bottom,#3B7DE9,#0054AC 100%);
  -webkit-transition: opacity 0.2s ease-in-out;
  transition: opacity 0.2s ease-in-out;
}

.c9.c9:after {
  content: "";
  z-index: -2;
  background: linear-gradient(to bottom,#0054AC,#3B7DE9);
  border-radius: 4px;
}

.c9.c9:hover:before {
  opacity: 0;
}

.c9.c9:active,
.c9.c9:focus,
.c9.c9:visited {
  outline: 0;
  box-shadow: 0 0 2px rgba(212,216,221,0.3);
}

.c5 {
  width: 100%;
  height: 34px;
  padding: 10px 5px;
  background-color: #FFFFFF;
  border-radius: 4px;
  border-width: 1px;
  border-style: solid;
  border-color: #D4D8DD;
  color: #333333;
  font-size: 13px;
  box-sizing: border-box;
}

.c5::-webkit-input-placeholder {
  color: #999999;
}

.c5::-moz-placeholder {
  color: #999999;
}

.c5:-ms-input-placeholder {
  color: #999999;
}

.c5::placeholder {
  color: #999999;
}

.c10 {
  background-color: #fff;
  border: 1px solid #D4D8DD;
  border-radius: 0 0 4px 4px;
  border-top: 0;
  color: #666;
  height: 32px;
  padding-left: 10px;
  padding-right: 10px;
  position: absolute;
  top: calc(100% - 1px);
  right: 20px;
}

.c10 svg {
  margin-left: 6px;
}

.c1 {
  width: 100%;
  padding-left: 20px;
  padding-right: 20px;
  background-color: #fff;
  border-bottom: 1px solid #D4D8DD;
  overflow-y: auto;
  -webkit-transition: max-height 199ms ease-in-out 0ms;
  transition: max-height 199ms ease-in-out 0ms;
  --max-row-number: 4;
  --height-actions: 64px;
  --height-filters-row: 48px;
  --height-filters-total-gaps: calc(15px * (var(--max-row-number) + 1));
  --height-filters: var(--height-filters-row) * var(--max-row-number) + var(--height-filters-total-gaps);
  max-height: calc(var(--height-actions) + var(--height-filters));
}

.c1[aria-hidden="true"] {
  border-bottom: none;
  visibility: hidden;
  max-height: 0;
  -webkit-transition: visibility 199ms linear 199ms,max-height 199ms ease-in-out 0ms;
  transition: visibility 199ms linear 199ms,max-height 199ms ease-in-out 0ms;
}

.c2 {
  padding-top: 15px;
  padding-bottom: 15px;
}

.c4 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-column-gap: 8px;
  column-gap: 8px;
}

.c3 + .c3 {
  margin-top: 15px;
}

.c6 {
  width: 100%;
  padding-top: 15px;
  padding-bottom: 15px;
  border-top: 1px solid #D4D8DD;
}

.c7 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-column-gap: 8px;
  column-gap: 8px;
}

.c7 > button {
  min-width: 76px;
}

.c0 {
  width: 100%;
  position: relative;
}

<div
    class="c0"
  >
    <div
      aria-hidden="false"
      class="c1"
    >
      <form>
        <div
          class="c2"
        >
          <div
            class="c3 c4"
          >
            <div
              class=""
            >
              <div>
                <label
                  for="text-field-1"
                >
                  Text label
                </label>
                <input
                  class="c5"
                  id="text-field-1"
                  type="text"
                />
              </div>
            </div>
          </div>
        </div>
        <div
          class="c6"
        >
          <div
            class="c7"
          >
            <button
              class="c8"
              color="settings"
            >
              クリア
            </button>
            <button
              class="c9"
              color="primary"
            >
              検索
            </button>
          </div>
        </div>
      </form>
    </div>
    <button
      aria-expanded="true"
      aria-haspopup="true"
      class="c10"
    >
      絞り込み条件 
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-caret-up fa-w-10 c11"
        data-icon="caret-up"
        data-prefix="fas"
        focusable="false"
        role="img"
        theme="[object Object]"
        viewBox="0 0 320 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M288.662 352H31.338c-17.818 0-26.741-21.543-14.142-34.142l128.662-128.662c7.81-7.81 20.474-7.81 28.284 0l128.662 128.662c12.6 12.599 3.676 34.142-14.142 34.142z"
          fill="currentColor"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
